fe/src/main/java/org/apache/doris/common/proc/SchemaChangeProcNode.java

```diff
@@ -42,6 +58,110 @@ public SchemaChangeProcNode(SchemaChangeHandler schemaChangeHandler, Database db
+        if (subExpr.getChild(1) instanceof StringLiteral && binaryPredicate.getOp() == BinaryPredicate.Operator.EQ) {
+            LOG.info("SubExpr value " + ((StringLiteral) subExpr.getChild(1)).getValue() + ", real value " + element);
+            return ((StringLiteral) subExpr.getChild(1)).getValue().equals(element);
+        }
+        if (subExpr.getChild(1) instanceof DateLiteral) {
+            Long leftVal = (new DateLiteral((String) element, Type.DATETIME)).getLongValue();
+            Long rightVal = ((DateLiteral) subExpr.getChild(1)).getLongValue();
+            LOG.info("Left value " + leftVal + ", right value " + rightVal);
+            switch( binaryPredicate.getOp()) {
+                case EQ:
+                case EQ_FOR_NULL:
+                    return leftVal == rightVal;
+                case GE:
+                    return leftVal >= rightVal;
+                case GT:
+                    return leftVal > rightVal;
+                case LE:
+                    return leftVal <= rightVal;
+                case LT:
+                    return leftVal < rightVal;
+                case NE:
+                    return leftVal != rightVal;
+                default:
+                    Preconditions.checkState(false, "No defined binary operator.");
+            }
+        }
+        return true;
+    }
+
+    public ProcResult fetchResultByFilter(HashMap<String, Expr> filter, ArrayList<OrderByPair> orderByPairs,
+                                          LimitElement limitElement) throws AnalysisException {
+        Preconditions.checkNotNull(db);
+        Preconditions.checkNotNull(schemaChangeHandler);
+
+        List<List<Comparable>> schemaChangeJobInfos = schemaChangeHandler.getAlterJobInfosByDb(db);
+
+        LOG.info("Begin fetch data.Filter size " + filter.size() );
+        for(Entry<String, Expr> entry : filter.entrySet()){
+            LOG.info("key " + entry.getKey() + ", value" + entry.getValue().getChild(1));
+        }
+
+        //where
+        List<List<Comparable>> jobInfos = new ArrayList<List<Comparable>>();
+        for (List<Comparable> infoStr : schemaChangeJobInfos) {
+            if (infoStr.size() != TITLE_NAMES.size()) {
+                LOG.warn("SchemaChangeJobInfos.size() " + schemaChangeJobInfos.size()
+                    + " not equal TITLE_NAMES.size() " + TITLE_NAMES.size());
+                continue;
+            }
+            List<Comparable> jobInfo = new ArrayList<Comparable>();
+            boolean isNeed = true;
+            for (int i = 0; i < infoStr.size(); i++) {
+                Comparable element = infoStr.get(i);
+                isNeed = filterResult(TITLE_NAMES.get(i), element, filter);
+                LOG.info("column "+ TITLE_NAMES.get(i) +", value " + element + ",isNeed " + isNeed);
+                if (!isNeed) {
+                    break;
+                }
+                jobInfo.add(element.toString());
```

Comment:
why not append infoStr to jobInfos when isNeed is true, other than putting element one by one